tools/ratbagctl.in

```diff
@@ -1373,18 +1373,18 @@ class RatbagParserRoot(object):
         self.parser = argparse.ArgumentParser(description="Inspect and modify a configurable device",
                                               add_help=False)
         self.parser.add_argument("-V", "--version", action="version", version="@version@")
-        self.parser.add_argument('--verbose', '-v', action='count', default=0)
         self.parser.add_argument('--help', '-h', action='store_true', default=False)
-        self.parser.add_argument('--nocommit', action='store_true', default=False)
-        if self.want_keepalive:
-            self.parser.add_argument('--keepalive', action='store_true', default=False)
-
         # retrieve the global options now and remove them from the processing
         ns, rest = self.parser.parse_known_args(input_string)
 
         if ns.help:
             return ns
 
+        self.parser.add_argument('--verbose', '-v', action='count', default=0)
+        self.parser.add_argument('--nocommit', action='store_true', default=False)
+        if self.want_keepalive:
+            self.parser.add_argument('--keepalive', action='store_true', default=False)
+
```

Comment:
I don't see how this is supposed to help here.

According to the commit message arguments should really be added to `command_parser` below, but I am still puzzled why this is needed.

The first parser reads the entire `input_string` and set the namespace `ns` accordingly. If `keepalive` is there is should be stored, no matter if the option comes in before or after the `device_or_list`.

Reply:
the code is basically
```
ns = parse_known_args()
....
ns = parse_known_args()
```
and the second `ns` is the one we're passing around - and that one doesn't have the flags from the first `ns` set anymore, including `keepalive`

Reply:
Oh right, then the correct fix should be:
```
        # retrieve the device and remove it from the command processing
        self.parser.add_argument('device_or_list', action="store")
        ns, rest = self.parser.parse_known_args(rest, namespace=ns)
```

We need to add the previous namespace to the new call

Reply:
yup, did that in the current version now